pasqal_cloud/__init__.py

```diff
@@ -251,28 +254,108 @@ def rebatch(
 
+    def get_jobs(
+        self,
+        job_ids: Union[List[UUID], EmptyFilter] = EmptyFilter(),
+        project_id: Union[List[UUID], EmptyFilter] = EmptyFilter(),
+        user_id: Union[List[str], EmptyFilter] = EmptyFilter(),
+        batch_id: Union[List[UUID], EmptyFilter] = EmptyFilter(),
+        status: Union[Sequence[JobStatus], EmptyFilter] = EmptyFilter(),
+        min_runs: Union[int, EmptyFilter] = EmptyFilter(),
+        max_runs: Union[int, EmptyFilter] = EmptyFilter(),
+        errors: Union[bool, EmptyFilter] = EmptyFilter(),
+        origin: Union[Sequence[JobOrigin], EmptyFilter] = EmptyFilter(),
+        start_date: Union[datetime, EmptyFilter] = EmptyFilter(),
+        end_date: Union[datetime, EmptyFilter] = EmptyFilter(),
+    ) -> List[Job]:
+        """
+        Retrieve a list of jobs matching filters.
+
+        Args:
+            job_ids: List of job UUIDs to retrieve.
+            project_id: List of project UUIDs to filter jobs by project.
+            user_id: List of user IDs to filter jobs by user.
+            batch_id: List of batch UUIDs to filter jobs by batch.
+            status: Status or list of statuses to filter jobs by status.
+            min_runs: Retrieve jobs with more or as many as this amount of runs.
+            max_runs: Retrieve jobs with less or as many as this amount of runs.
+            errors: Boolean to filter jobs that have errors.
+            origin: Origin or list of origins to filter jobs by origin.
+            start_date: Retrieve jobs created at or after this datetime.
+            end_date: Retrieve jobs created at or before this datetime.
+
+        Returns:
+            List[Job]: A list of Jobs matching the filters.
+
+        Raises:
+            JobFetchingError: If fetching jobs call failed.
+        """
+        query_params: Dict[str, Any] = {}
+        if not isinstance(job_ids, EmptyFilter):
+            query_params["id"] = job_ids
+        if not isinstance(project_id, EmptyFilter):
+            query_params["project_id"] = project_id
+        if not isinstance(user_id, EmptyFilter):
+            query_params["user_id"] = user_id
+        if not isinstance(batch_id, EmptyFilter):
+            query_params["batch_id"] = batch_id
+        if not isinstance(status, EmptyFilter):
+            query_params["status"] = status
+        if not isinstance(min_runs, EmptyFilter):
+            query_params["min_runs"] = min_runs
+        if not isinstance(max_runs, EmptyFilter):
+            query_params["max_runs"] = max_runs
+        if not isinstance(errors, EmptyFilter):
+            query_params["errors"] = errors
+        if not isinstance(origin, EmptyFilter):
+            query_params["origin"] = origin
+        if not isinstance(start_date, EmptyFilter):
+            query_params["start_date"] = start_date
+        if not isinstance(end_date, EmptyFilter):
+            query_params["end_date"] = end_date
+
+        params = JobFilterParams(**query_params)
```

Comment:
How about adding a dedicated method to `JobFilterParams` called `from_user_inputs` which gets rid of all the EmptyFilter (and do the casting if necessary) and returns the `JobFilterParams` object.

Benefits:
- less clutter in all the SDK methods
- centralize all the operations in the JobFilterParams. If we add a field it's going to be easier to remember to update that method
- Better type validation. Here there is no type validation when we do `JobFilterParams(**query_params)` because we go through a dictionnary. With a method we can type all the arguments.